arrow-continuations.api

```diff
@@ -1,123 +1,45 @@
-	public fun <init> (Lkotlin/jvm/functions/Function2;)V
-	public final fun getF ()Lkotlin/jvm/functions/Function2;
-	public final fun invoke ()Ljava/lang/Object;
-	public fun reset (Lkotlin/jvm/functions/Function2;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public fun shift (Lkotlin/jvm/functions/Function3;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public fun shiftCPS (Lkotlin/jvm/functions/Function2;Lkotlin/jvm/functions/Function3;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
+public final class arrow/continuations/Effect$Companion {
+	public final fun restricted (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
+	public final fun suspended (Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function1;Lkotlin/jvm/functions/Function2;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 }
 
-public final class arrow/continuations/generic/DelimContScope$CPSCont : arrow/continuations/generic/DelimitedContinuation {
-	public fun <init> (Lkotlin/jvm/functions/Function3;)V
-	public final fun copy (Lkotlin/jvm/functions/Function3;)Larrow/continuations/generic/DelimContScope$CPSCont;
-	public static synthetic fun copy$default (Larrow/continuations/generic/DelimContScope$CPSCont;Lkotlin/jvm/functions/Function3;ILjava/lang/Object;)Larrow/continuations/generic/DelimContScope$CPSCont;
-	public fun equals (Ljava/lang/Object;)Z
-	public fun hashCode ()I
-	public fun invoke (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public fun toString ()Ljava/lang/String;
+public final class arrow/continuations/Reset {
+	public static final field INSTANCE Larrow/continuations/Reset;
+	public final fun restricted (Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
+	public final fun suspended (Lkotlin/jvm/functions/Function2;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
 }
 
-public final class arrow/continuations/generic/DelimContScope$Companion {
-	public final fun reset (Lkotlin/jvm/functions/Function2;)Ljava/lang/Object;
-}
-
-public final class arrow/continuations/generic/DelimContScope$SingleShotCont : arrow/continuations/generic/DelimitedContinuation {
-	public fun <init> (Lkotlin/coroutines/Continuation;Ljava/util/List;)V
-	public final fun copy (Lkotlin/coroutines/Continuation;Ljava/util/List;)Larrow/continuations/generic/DelimContScope$SingleShotCont;
-	public static synthetic fun copy$default (Larrow/continuations/generic/DelimContScope$SingleShotCont;Lkotlin/coroutines/Continuation;Ljava/util/List;ILjava/lang/Object;)Larrow/continuations/generic/DelimContScope$SingleShotCont;
-	public fun equals (Ljava/lang/Object;)Z
-	public fun hashCode ()I
-	public fun invoke (Ljava/lang/Object;Lkotlin/coroutines/Continuation;)Ljava/lang/Object;
-	public fun toString ()Ljava/lang/String;
+public class arrow/continuations/generic/ControlThrowable : java/lang/Throwable {
+	public fun <init> ()V
+	public fun fillInStackTrace ()Ljava/lang/Throwable;
```

Comment:
Wondering if this stacktrace is actually useful in the context of structured concurrency if someone has a massive `either` block nested

Reply:
Note: we can introduce `arrow.fillstacktrace=enabled` to generate stacktraces for all currently no-op stacktraces later.